lib/ansible/modules/cloud/amazon/ecs_attribute.py

```diff
@@ -0,0 +1,264 @@
+  delegate_to: localhost
+
+# Delete attribute
+- ecs_attribute:
+    cluster: test-cluster
+    name: flavor
+    state: absent
+    ec2_instance_id: "{{ ec2_id }}"
+  delegate_to: localhost
+'''
+
+RETURN = '''
+attributes:
+    description: list of attributes
+    type: list of objects
+    contains:
+        name:
+            description: name of the attribute
+            type: string
+        value:
+            description: value of the attribute
+            returned: if present
+            type: string
+        cluster:
+            description: cluster name
+            type: string
+        ec2_instance_id:
+            description: ec2 instance id of ecs container instance
+            type: string
+'''
+
+try:
+    import boto
+    import botocore
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+class EcsAttribute:
+    """Handles ECS Cluster Attribute"""
+
+    def __init__(self, module, cluster, name, value=None):
+        self.module = module
+        self.cluster = cluster
+        self.name = name
+        self.value = value
+
+        try:
+            region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+            if not region:
+                module.fail_json(msg=("Region must be specified as a parameter,"
+                                      "in EC2_REGION or AWS_REGION environment"
+                                      "variables or in boto configuration file"))
+            self.ecs = boto3_conn(module, conn_type='client', resource='ecs',
+                    region=region, endpoint=ec2_url, **aws_connect_kwargs)
+        except boto.exception.NoAuthHandlerFound as e:
+            self.module.fail_json(msg="Can't authorize connection - %s" % str(e))
+
+
+    def _ec2_id_to_ecs_arn(self, ec2_id):
+        try:
+            ecs_instances_arns = self.ecs.list_container_instances(cluster=
+                    self.cluster)['containerInstanceArns']
+            ec2_instances = self.ecs.describe_container_instances(cluster=self.cluster,
+                    containerInstances=ecs_instances_arns)['containerInstances']
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(msg="Can't connet to the cluster - %s" % str(e))
+
+        try:
+            ecs_arn = next(i for i in ec2_instances if i['ec2InstanceId'] == ec2_id)
+        except StopIteration as e:
+            self.module.fail_json(msg="EC2 instance Id not found in ECS cluster" % (str(e)))
+
+        return ecs_arn['containerInstanceArn']
+
+
+    def _get_attr_obj(self, ec2_id, skip_value=False):
+
+        attr_obj = {
+                'targetType': 'container-instance',
+                'targetId': self._ec2_id_to_ecs_arn(ec2_id),
```

Comment:
so I haven't tried the API but just wondering if we could do things more efficiently. 
http://boto3.readthedocs.io/en/latest/reference/services/ecs.html#ECS.Client.put_attributes 

For `targetId` it says that if you use `targetType` as `container-instance` you can use the short form ID for a resource instead. Therefore is that doable to do `targetId: ec2_id` ?